### ISSUE TITLE
Add option --no-keepalive (curl option)

### ISSUE THREAD

**Issue by fabricereix** [Orange-OpenSource/hurl#3253]
By default, Hurl keeps connections alive between requests (like curl).
It might be useful to disable it.

I've tested an API endpoint that didn't return any content (HTTP 204).
In order to get the response content (HTTP 200), I needed to disable the keep-alive (and force HTTP1.1 over HTTP2)

**Comment by fabricereix** [comment]
In fact, I needed to add a delay to get a correct response from my API endpoint.
This option might still be useful but clearly low priority.

**Comment by theoforger** [comment]
Hello @jcamiel ! I'd love to work on this feature!

I assume [`curl::easy::Easy::tcp_keepalive`](https://docs.rs/curl/latest/curl/easy/struct.Easy2.html#method.tcp_keepalive) is what I should use? But according to the doc it already defaults to `false`. Am I missing anything here?

Thanks!

**Comment by jcamiel** [comment]
@theoforger I've the impression that the right option to disable reusing connection is not [CURLOPT_TCP_KEEPALIVE](https://curl.se/libcurl/c/CURLOPT_TCP_KEEPALIVE.html), but something like [CURLOPT_FRESH_CONNECT](https://curl.se/libcurl/c/CURLOPT_FRESH_CONNECT.html) or [CURLOPT_FORBID_REUSE](https://curl.se/libcurl/c/CURLOPT_FORBID_REUSE.html). I've to dig to understand the difference but `CURLOPT_TCP_KEEPALIVE` is more about sending tcp keepalive probe and is independent on connection reuse or not.

**Comment by theoforger** [comment]
@jcamiel Thanks for the info!

I did some testing on all three options:

```rust
    let mut easy = Easy::new();
    easy.url("http://127.0.0.1:8080")?;

    easy.fresh_connect(false)?; // CURLOPT_FRESH_CONNECT
    easy.tcp_keepalive(true)?; // CURLOPT_TCP_KEEPALIVE
    easy.forbid_reuse(false)?; // CURLOPT_FORBID_REUS

    easy.perform()?;
    println!("Response code: {}", easy.response_code()?);
```
while monitoring the connections with `ss -tnp`.

For some reasons no matter what options I use, libcurl doesn't keep any connection open afterwards. What's happening here?

**Comment by theoforger** [comment]
You're right. It's working now!

My understanding is that:
- [CURLOPT_FRESH_CONNECT](https://curl.se/libcurl/c/CURLOPT_FRESH_CONNECT.html) makes sure a new connection is created for each request
- [CURLOPT_FORBID_REUSE](https://curl.se/libcurl/c/CURLOPT_FORBID_REUSE.html) makes sure connections are closed after transfer

Either of them can stop reuse. Do we want to combine them or keep them as two separate flags?

Also is there still an interest for `--no-keepalive`?

**Comment by jcamiel** [comment]
Don't think there is a need to `keepalive` / `no-keepalive`. We can begin to implement `CURLOPT_FRESH_CONNECT` (`--fresh-connect`) and that should be sufficient to expose an option to be able to not reuse connection.

**Comment by jcamiel** [comment]
@fabricereix one thing to note it that there is no curl cli equivalent of `CURLOPT_FRESH_CONNECT`, it's just a libcurl option.

**Comment by theoforger** [comment]
Hello again! I just filed a PR. I mainly used #2468 as a reference to work on this. Let me know if there's any changes you wish to make 🙏

**Comment by fabricereix** [comment]
This `--no-keepalive` curl option does not seem in fact straightforward.
We would need a test for which you could see a difference with and without this option, so that we could also validate with Hurl.
Contrary to what I thought initially, it does not seem to be directly related to reusing connection.

**Comment by theoforger** [comment]
@jcamiel I would still love to contribute though. If there's anything I can work on, feel free to let me know! 🙏

**Comment by jcamiel** [comment]
@theoforger Of course, let us just check something that will be merged without problem, so we don't waste your time this time!
